OpenRA.Mods.Common/Traits/Demolishable.cs

```diff
@@ -38,14 +45,35 @@ public bool PreventsEjectOnDeath(Actor self)
 			return info.PreventsEjectOnDeath;
 		}
 
-		public void Demolish(Actor self, Actor saboteur)
+		public void Demolish(Actor self, Actor saboteur, int delay)
 		{
-			self.Kill(saboteur);
+			triggerDemolition = true;
+			this.delay = delay;
+			this.saboteur = saboteur;
+
+			modifiers = self.TraitsImplementing<IDamageModifier>()
+				.Concat(saboteur.Owner.PlayerActor.TraitsImplementing<IDamageModifier>())
+				.Select(t => t.GetDamageModifier(saboteur, null));
 		}
 
 		public bool IsValidTarget(Actor self, Actor saboteur)
 		{
 			return true;
 		}
+
+		void ITick.Tick(Actor self)
+		{
+			if (self.IsDead)
+				return;
+
+			if (!triggerDemolition)
+				return;
+
+			if (--delay > 0)
```

Comment:
If you change this to `if (delay == 0 || --delay > 0)` then you can remove the `triggerDemolition` bool.